### DISCUSSION TITLE
Selecting the optimal items for users to rate in order to profile them with the least amount of effort on their part

### DISCUSSION THREAD

**Discussion by vibl** [online-ml/river#1141] [General]
Thank you for this amazing lib! I wish I had found it earlier.

I have a problem I need to solve and I wonder if this is the right lib for it.

I have an app where users rate pictures spontaneously and then they get matched with similar users. I want to present specific pictures that they should rate in order to profile them better (so that I can match them better later). I want to present the fewest possible pictures to rate, in order to minimize their work. In other words, each picture that I ask them to rate should maximize the information I gain on them.

Can I do that with River?

**Comment by MaxHalford** [comment]
Hey there @vibl! It sounds like an interesting problem. However, it seems to me like it still needs to be framed a bit more. There's nothing in River that could be used as is. In fact, because there isn't a clear solution, I would even suggest taking a batch approach. You're probably going to complicate your life by doing a streaming approach.
